### PR TITLE
fix: pass --extra-meta for rattler-build

### PR DESCRIPTION
**Pull request by nichmor**
Checklist
* [ ] Added a ``news`` entry
* [ ] Regenerated schema JSON if schema altered (`python conda_smithy/schema.py`)

Closes https://github.com/conda-forge/conda-smithy/issues/2034

Actual run can be seen here: https://github.com/conda-forge/jolt-physics-feedstock/pull/2

### CODE REVIEW

**Review by beckermr — CHANGES_REQUESTED** [review]
Does rattler not support the --no-test option? We need that for some cross-compile cases when emulation is turned off or we are working with osx-arm on osx-64.

**Comment by baszalmstra** [comment]
There is [a `--no-test`](https://prefix-dev.github.io/rattler-build/latest/reference/cli/#modifying-result) for rattler-build but Im not entirely sure if it does the same thing as conda-build. Can you check @nichmor ?

**Comment by nichmor** [comment]
should also fix: https://github.com/conda-forge/crane-feedstock/pull/6

**Comment by baszalmstra** [comment]
@wolfv Can you verify that `--no-test` for `rattler-build` do the same thing as `--no-test` for `conda-build`?

**Comment by wolfv** [comment]
I did just double check, and rattler-build unconditionally skips test execution which is what conda-build also does.

File: conda_smithy/templates/build_steps.sh.tmpl [hunk 1]
```diff
@@ -76,11 +76,11 @@ ulimit -n 1024
 # make the build number clobber
 make_build_number "${FEEDSTOCK_ROOT}" "${RECIPE_ROOT}" "${CONFIG_FILE}"
 
-{% if test == "native" -%}
+{% if test == "native" and conda_build_tool != "rattler-build" -%}
 if [[ "${HOST_PLATFORM}" != "${BUILD_PLATFORM}" ]] && [[ "${BUILD_WITH_CONDA_DEBUG:-0}" != 1 ]]; then
     EXTRA_CB_OPTIONS="${EXTRA_CB_OPTIONS:-} --no-test"
 fi
-{%- elif test == "native_and_emulated" -%}
+{%- elif test == "native_and_emulated" and conda_build_tool != "rattler-build" -%}
```
Comment (wolfv) [inline conda_smithy/templates/build_steps.sh.tmpl:83]: is this correct? If we need `--no-test` then this is going to be missing in the rattler-build case, right?

Comment (nichmor) [reply conda_smithy/templates/build_steps.sh.tmpl:83]: Initially it was disabled by me, but now that `--no-test` is working for rattler-build I need to remove this check

Comment (beckermr) [reply conda_smithy/templates/build_steps.sh.tmpl:83]: I agree this looks wrong.

Comment (baszalmstra) [reply conda_smithy/templates/build_steps.sh.tmpl]: I reverted these changes.

**Review by wolfv — APPROVED** [review]

**Comment by wolfv** [comment]
Do we need to set the rattler-build version somewhere to `>=0.21.0`?

Looks good to me!

**Comment by nichmor** [comment]
> Do we need to set the rattler-build version somewhere to `>=0.21.0`?
> 
> Looks good to me!

I think the minimal should be `>=0.20.0` as it was released there, and we need to set it here: https://github.com/conda-forge/rattler-build-conda-compat-feedstock/blob/main/recipe/meta.yaml#L24.

but we can merge this anyway,no?

**Comment by wolfv** [comment]
Yep!

**Pull request merged**